### PR TITLE
feat(core/application): Add 'autoActivate' toggle for DataSources

### PR DESCRIPTION
**Pull request by christopherthielen**
- Also deprecate and remove usage of `DataSourceConfig` because it doesn't do anything useful

@anotherchrisberry @icfantv @jrsquared ptal

### CODE REVIEW

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts [hunk 1]
```diff
@@ -329,6 +291,11 @@ export class ApplicationDataSource {
     if (!config.activeState) {
       this.activeState = '**' + this.sref + '.**';
     }
+
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/application/service/applicationDataSource.ts]: Large diffs are not rendered by default

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts [hunk 2]
```diff
@@ -329,6 +291,11 @@ export class ApplicationDataSource {
...
+    if (config.autoActivate) {
+      $uiRouter.transitionService.onSuccess({ entering: this.activeState }, () => this.activate());
+      $uiRouter.transitionService.onSuccess({ exiting: this.activeState }, () => this.deactivate());
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/application/service/applicationDataSource.ts]: Large diffs are not rendered by default

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts [hunk 3]
```diff
@@ -329,6 +291,11 @@ export class ApplicationDataSource {
...
+    }
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/application/service/applicationDataSource.ts]: Large diffs are not rendered by default

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts [hunk 4]
```diff
@@ -1,102 +1,76 @@
...
    */
-  public sref: string;
+  afterLoad?: (application: Application) => void;
 
   /**
-   * (Optional) whether this data source should be included in the application by default
-   *
-   * Use this for optional or experimental features that users must explicitly enable.
+   * (Optional) Used to determine when the application header tab should appear active; the field will be used via
+   * $state.includes(activeState).
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/service/applicationDataSource.ts]: This comment is really bad

File: app/scripts/modules/core/src/application/service/applicationDataSource.ts (hunk 3 above)
Comment (anotherchrisberry) [reply app/scripts/modules/core/src/application/service/applicationDataSource.ts]: A+ nice work

**Comment by anotherchrisberry** [comment]
real nice

**Pull request merged**